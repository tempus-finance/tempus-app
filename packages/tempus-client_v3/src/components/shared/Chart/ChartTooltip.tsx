import { ReactElement, ReactNode } from 'react';
import { TooltipProps } from 'recharts';
import { NameType, ValueType } from 'recharts/types/component/DefaultTooltipContent';
import Tooltip from '../Tooltip';

interface ChartTooltipProps<X extends NameType, Y extends ValueType> {
  tooltipContent: (x: X, y: Y) => ReactNode;
}

function ChartTooltip<X extends NameType, Y extends ValueType>(
  props: ChartTooltipProps<X, Y> & TooltipProps<Y, X>,
): ReactElement<ChartTooltipProps<X, Y> & TooltipProps<Y, X>> {
  const { tooltipContent, coordinate, viewBox, payload } = props;
  return (
    <div className="tc__chart__tooltip-chart">
      <Tooltip
        open
        placement={
          coordinate && coordinate.x && viewBox && viewBox.width && coordinate.x / viewBox.width > 0.5
            ? 'left'
            : 'right'
        }
      >
        {payload && payload[0] !== undefined && tooltipContent(payload[0].payload.x, payload[0].payload.y)}
      </Tooltip>
    </div>
  );
}

export default ChartTooltip;
