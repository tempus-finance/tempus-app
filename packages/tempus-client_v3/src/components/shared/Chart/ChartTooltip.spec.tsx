import { render } from '@testing-library/react';
import ChartTooltip, { ChartTooltipProps } from './ChartTooltip';

const defaultProps: ChartTooltipProps = {
  tooltipContent: (x: number, y: number) => (
    <div style={{ width: 300, height: 300 }}>
      X: {x}
      Y: {y}
    </div>
  ),
  payload: [
    {
      payload: {
        x: 100,
        y: 200,
      },
    },
  ],
};

const subject = (props: ChartTooltipProps) => render(<ChartTooltip {...props} />);

describe('ChartTooltip', () => {
  it('renders a chart tooltip with left placement', () => {
    const { container } = subject({ ...defaultProps, coordinate: { x: 100 }, viewBox: { width: 150 } });

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('renders a chart tooltip with right placement', () => {
    const { container } = subject({ ...defaultProps, coordinate: { x: 10 }, viewBox: { width: 150 } });

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
