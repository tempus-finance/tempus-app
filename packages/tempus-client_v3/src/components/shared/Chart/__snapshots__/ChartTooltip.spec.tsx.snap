// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartTooltipContent renders a chart tooltip with left placement 1`] = `
<div>
  <div
    class="tc__chart__tooltip-chart"
  >
    <div
      class="tc__tooltip tc__tooltip__left"
    >
      <div
        style="width: 300px; height: 300px;"
      >
        X: 
        100
        Y: 
        200
      </div>
    </div>
  </div>
</div>
`;

exports[`ChartTooltipContent renders a chart tooltip with right placement 1`] = `
<div>
  <div
    class="tc__chart__tooltip-chart"
  >
    <div
      class="tc__tooltip tc__tooltip__right"
    >
      <div
        style="width: 300px; height: 300px;"
      >
        X: 
        100
        Y: 
        200
      </div>
    </div>
  </div>
</div>
`;
